src/components/study-pane.js

```diff
@@ -175,6 +176,23 @@ const StudyPane = () => {
         history.replace("/studies/" + studyName)
     }
 
+    const handleUpdateSwitchState = useCallback( (studyName, breakerId, open, clickedDomElement, breakerMetadata) => {
+        updateSwitchState(studyName, breakerId, open).then( response => response.ok ? updateSvgSwitch(clickedDomElement, open, breakerMetadata) : console.debug(response));
+    }, []);
+
+    function updateSvgSwitch(clickedDomElement, open, breakerMetadata) {
```

Comment:
This is not the right way to do it as for instance when topological coloring will be activated, update can only be done on server side. So we need to reload again from the server the SVG generated with the new topology.